google-cloud-storage/src/main/java/com/google/cloud/storage/spi/v1/HttpStorageRpc.java

```diff
@@ -750,10 +751,11 @@ public long read(
       } else {
         req.setReturnRawInputStream(false);
       }
-      req.getMediaHttpDownloader().setBytesDownloaded(position);
-      req.getMediaHttpDownloader().setDirectDownloadEnabled(true);
-      req.executeMedia().download(outputStream);
-      return req.getMediaHttpDownloader().getNumBytesDownloaded();
+      MediaHttpDownloader mediaHttpDownloader = req.getMediaHttpDownloader();
+      mediaHttpDownloader.setBytesDownloaded(position);
+      mediaHttpDownloader.setDirectDownloadEnabled(true);
+      req.executeMediaAndDownloadTo(outputStream);
+      return mediaHttpDownloader.getNumBytesDownloaded();
```

Comment:
Is this a refactor or is there magical Apiary functionality hidden in this?

Reply:
I have since moved away from the previous executeAndDownload method change. This was not properly respecting `returnRawInputStream` which would break gzip automatic transcoding if you wanted to actually download the zipped bytes. Instead, we are now setting the range header ourselves.